src/R/Support/Impl/Help/Packages/PackageIndex.cs

```diff
@@ -201,38 +200,55 @@ public sealed class PackageIndex : IPackageIndex {
             }
         }
 
+        private async Task UpdateInstalledPackagesAsync() {
+            var token = await _buildIndexLock.WaitAsync();
+            try {
+                var installed = await GetInstalledPackagesAsync();
+                var installedNames = installed.Select(p => p.Package).Append("rtvs");
+
+                var currentNames = _packages.Keys.ToArray();
+                var removedNames = currentNames.Except(installedNames);
+                _packages.RemoveWhere((kvp) => removedNames.Contains(kvp.Key));
+
+                var added = installed.Where(p => !currentNames.Contains(p.Package));
+                await AddPackagesToIndexAsync(added);
+            } catch (OperationCanceledException) { }
+        }
+
         /// <summary>
         /// From the supplied names selects packages that are not in the index and attempts
         /// to add them to the index. This typically applies to packages that were just installed.
         /// </summary>
         private async Task<IEnumerable<IPackageInfo>> TryAddMissingPackagesAsync(IEnumerable<string> packageNames) {
-            var list = new List<IPackageInfo>();
+            var info = Enumerable.Empty<IPackageInfo>();
             // Do not attempt to add new package when index is still being built
             if (packageNames.Any() && _buildIndexLock.IsSet) {
                 try {
                     var installedPackages = await GetInstalledPackagesAsync();
                     var packagesNotInIndex = installedPackages.Where(p => packageNames.Contains(p.Package));
-                    foreach (var p in packagesNotInIndex) {
-                        var info = new PackageInfo(_host, p.Package, p.Description, p.Version);
-                        _packages[p.Package] = info;
+                    info = await AddPackagesToIndexAsync(packagesNotInIndex);
+                } catch (RHostDisconnectedException) { }
+            }
+            return info;
+        }
 
-                        await info.LoadFunctionsIndexAsync();
-                        _functionIndex.RegisterPackageFunctions(info);
+        private async Task<IEnumerable<IPackageInfo>> AddPackagesToIndexAsync(IEnumerable<RPackage> packages) {
+            var list = new List<IPackageInfo>();
```

Comment:
Doesn't matter much. Would you rather prefer `yield return'? I just don't want to change logic too much at this point.